assets/src/edit-story/components/panels/link.js

```diff
@@ -82,29 +85,43 @@ function LinkPanel({ selectedElements, onSetProperties }) {
   );
   const handleSubmit = useCallback(
     (evt) => {
-      onSetProperties({ link: state?.url ? state : null });
+      onSetProperties({ link: state?.url ? { ...state } : null });
       if (evt) {
         evt.preventDefault();
       }
     },
     [state, onSetProperties]
   );
 
+  const {
+    actions: { getLinkMetadata },
+  } = useAPI();
+
   const canLink = selectedElements.length === 1 && !isFill;
-  const populateMetadata = useCallback(
-    debounce(300, async (/** url */) => {
-      // TODO(wassgha): Implement getting the page metadata
-    })
-  );
+
+  const [populateMetadata] = useDebouncedCallback((url) => {
+    if (!validUrl.isUri(url)) {
+      return;
+    }
+    setFetchingMetadata(true);
+    getLinkMetadata(url).then(({ title, image }) => {
```

Comment:
Let's also immediately assume that URL itself can change. E.g. "http://" can be switched to "https://".

Reply:
We don't get that information from the current API, I'll make an issue for @swissspidy  to look into

Reply:
@jauyong could you please triage this and make an issue accordingly? The task is to return the redirected url from the PHP API for retrieving url metadata.

Reply:
There are some nice utils for [urls](https://github.com/WordPress/gutenberg/tree/master/packages/url/src) in gutenberg - 
Might be useful here. Specially [prepend-http](https://github.com/WordPress/gutenberg/blob/master/packages/url/src/prepend-http.js)